multi_company_remove/models/res_company.py

```diff
@@ -141,24 +141,35 @@ def move_users(self, remaining_company):
             SET company_id = %s
             WHERE company_id IN %s
         """
-        cr.execute(statement, (remaining_company.id, self.ids))
-        user_model = self.env['res.users']
-        users_to_change = user_model.search([
-            ('company_id', 'in', self.ids),
-        ])
-        if users_to_change:
-            for user in users_to_change:
-                if remaining_company.id not in user.company_ids.ids:
-                    user.write({'company_ids': [(4, remaining_company.id)]})
-            users_to_change.write({'company_id': remaining_company.id})
-        # No partner if it is a user should belong to company to be deleted:
-        users_to_change = user_model.search([])
-        for user in users_to_change:
-            if user.partner_id.company_id and \
-                    user.partner_id.company_id.id in self.ids:
-                user.partner_id.write({
-                    'company_id': remaining_company.id,
-                })
+        cr.execute(statement, (remaining_company.id, tuple(self.ids)))
+
+        # select the users that belong in in the current company
+        # and they are connected with the remaining_company
+        # move them to the remaining_company.
+        # and change their partner_id.company_id to remaining_company
+        statement_select_users = """
+        SELECT id FROM res_users
+            INNER JOIN res_company_users_rel ON res_users.id = res_company_users_rel.user_id
+            WHERE res_users.company_id=ANY(%s) AND
+            res_company_users_rel.cid = %s
+        """
+        cr.execute(statement_select_users, (self.ids,
```

Comment:
If it does not fit on one line, just starts parms on the next line indented by four spaces.